code/WorkInProgress/nadir_antenna.dm

```diff
@@ -165,34 +210,285 @@ var/global/obj/machinery/communications_dish/transception/transception_array
+					//this bit is so you don't spend more on charge than you actually need to charge the cell
+					if(intcap.charge + yield_to_cell > src.intcap.maxcharge)
+						yield_to_cell = src.intcap.maxcharge - src.intcap.charge
+						final_draw = yield_to_cell * 500
+					src.intcap.give(yield_to_cell)
+					powernet.newload += final_draw
+					var/area/arrayarea = get_area(src) //gotta let the grid know!
+					arrayarea.use_power(final_draw,EQUIP)
+			else
+				src.intcap_charging = FALSE
+		else
+			src.intcap_charging = FALSE
+
+	///Layer for using internal capacitor; separated to intercept rigged cells and handle with custom damage behavior
+	proc/use_intcap(var/use_amount)
+		. = TRUE
+		if(src.intcap.rigged)
+			intcap_failure()
+			. = FALSE
+		else
+			src.intcap.use(use_amount)
+
+	///Sabotaged cells, instead of blowing out the turf, will blow out the associated microvoltage cabinet and bring the array offline
+	proc/intcap_failure()
+		src.intcap.rigged = FALSE
+		src.intcap_charging = FALSE
+		src.status |= BROKEN
+		src.primed = FALSE
+		src.intcap_door_open = FALSE
+		src.repair_status = 0
+		src.UpdateIcon()
+		if (intcap.rigger)
+			message_admins("[key_name(intcap.rigger)]'s rigged cell damaged the transception array at [log_loc(src)].")
+			logTheThing(LOG_COMBAT, intcap.rigger, "'s rigged cell damaged the transception array at [log_loc(src)].")
+
+		src.visible_message("<span class='alert'><b>[src]'s internal capacitor compartment explodes!</b></span>")
+
+		for(var/client/C in clients)
+			playsound(C.mob, 'sound/effects/explosionfar.ogg', 35, 0)
+
+		var/epicenter = get_turf(src)
+		playsound(epicenter, "explosion", 90, 1)
+		//this doesn't actually explode because turf safe explosions don't respect "space" (ocean) turfs, and I'd like surroundings not punctured
+
+		SPAWN(0)
+			qdel(src.intcap)
+			src.intcap = null
+
+	ex_act(severity)
+		return //it's a tough critter if you're not damaging it from the inside
+
+/obj/machinery/communications_dish/transception/attack_hand(mob/user)
+	if(src.intcap && intcap_door_open)
+		boutput(user, "<span class='notice'>You remove \the [intcap] from the cabinet's cell compartment.</span>")
+		playsound(src, 'sound/items/Deconstruct.ogg', 40, 1)
+
+		user.put_in_hand_or_drop(src.intcap)
+		src.intcap = null
 		return
+	..()
+
+/obj/machinery/communications_dish/transception/attackby(obj/item/I, mob/user)
+	src.add_fingerprint(user)
+	if(status & BROKEN && !istype(I, /obj/item/grab/))
+		var/tell_you_what_to_do_next = TRUE //probably-simpler way to tell people what to do next
+
+		if (isweldingtool(I))
+			if (src.repair_status <= 2)
+				boutput(user, "You start repairing the damaged sections of the outer cabinet plating.")
+				actions.start(new/datum/action/bar/icon/array_repair_weld(user,I,src), user)
+				tell_you_what_to_do_next = FALSE
+
+		else if (iswrenchingtool(I))
+			if (src.repair_status == 3 || src.repair_status == 7)
+				var/cursed_check = src.repair_status - 3
```

Comment:
this is evil and also kinda bad to hard code because it's dependant on the wrench steps having those exact values. Just do the defines thing I requested above and then 
`"You start [src.repair_status = TRSC_REMOVE_BOLTS ? "removing" : "reinstalling"] the ...`
or whatever

Reply:
Exorcised.